pkg/catcher/catcher.go

```diff
@@ -0,0 +1,56 @@
+// package catcher is a convient wrapper to interconnect signals with
+// contexts, etc
+package catcher
+
+import (
+	"context"
+	"os"
+	"os/signal"
+)
+
+type Catcher interface {
+	Catch(ctx context.Context) context.Context
+}
+
+type SignalCatcher struct {
+	sigs     []os.Signal
+	oncaught func(os.Signal)
+}
+
+func NewSignalCatcher(sigs ...os.Signal) *SignalCatcher {
+	return &SignalCatcher{sigs: sigs}
+}
+
+func (s *SignalCatcher) SetOnCaught(cb func(os.Signal)) {
+	s.oncaught = cb
+}
+
+func (s *SignalCatcher) Catch(ctx context.Context) context.Context {
+	//XXX this could be made more efficient by having the signal handler
+	// be a parent of each context created and keeping just one signal
+	// channel around.  then when we get the signal, we cancel the parent
+	// context and all the children get canceled.
+	// but for now, this is ok
```

Comment:
Please remove; I don't think it's helping to explain, as its describing how this is essentially being used now.